src/auth/oauth2client.ts

```diff
@@ -660,20 +679,34 @@ export class OAuth2Client extends AuthClient {
     options: GetTokenOptions
   ): Promise<GetTokenResponse> {
     const url = this.endpoints.oauth2TokenUrl.toString();
+    const headers: Headers = {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    };
     const values = {
```

Comment:
You can add `grant_type` to the `GetTokenOptions` type.

```suggestion
    const values: Omit<GetTokenOptions, 'codeVerifier'> = {
```